src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/openapi/ClientsApiController.java

```diff
@@ -406,6 +404,35 @@ private ClientType getClientType(String encodedId) {
         return new ResponseEntity<>(subjects, HttpStatus.OK);
     }
 
+    @Override
+    @PreAuthorize("hasAuthority('SEND_CLIENT_REG_REQ')")
+    public ResponseEntity<Void> registerClient(String encodedClientId) {
+        ClientId clientId = clientConverter.convertId(encodedClientId);
+        try {
+            clientService.registerClient(clientId);
+        } catch (GlobalConfOutdatedException e) {
+            throw new BadRequestException(e);
+        } catch (ClientNotFoundException e) {
+            throw new ResourceNotFoundException(e);
+        }
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+    }
+
+    @Override
+    @PreAuthorize("hasAuthority('SEND_CLIENT_DEL_REQ')")
+    public ResponseEntity<Void> unregisterClient(String encodedClientId) {
+        ClientId clientId = clientConverter.convertId(encodedClientId);
+        try {
+            clientService.unregisterClient(clientId);
+        } catch (GlobalConfOutdatedException | ClientService.CannotUnregisterOwnerException
+                | ActionNotPossibleException e) {
```

Comment:
`ActionNotPossibleException` and `CannotUnregisterOwnerException` are now considered a conflict since the request itself is not 'bad'. This would also apply to the `GlobalConfOutdatedException` but let's still follow the current approach that we have and treat it as a bad request (the `GlobalConfOutdatedException` case has been discussed sometime before IIRC (?))

Reply:
Yeah, `GlobalConfOutdatedException` is a bit suspect too, but since it is widely used convention let's not touch it in this ticket.

Reply:
There was some discussion about it in some ticket, unfortunately can't remember where...